src/library/kernels/atomic_reduce.cl

```diff
@@ -57,13 +76,13 @@ void atomic_add_float (global VALUE_TYPE *ptr, VALUE_TYPE temp)
         {
                 prevVal = as_uint(*ptr);
                 newVal = as_uint(temp + *ptr);
-        } while (atomic_cmpxchg((global unsigned int *)ptr, prevVal, newVal) != prevVal);
+        } while (atomic_cmpxchg((global unsigned ing *)ptr, prevVal, newVal) != prevVal);
```

Comment:
Good catch, sorry about that.

I'll fix that and squash these three changes together. I'll leave this open for a bit longer to see if there are any other comments, but I'll cancel and submit a new PR later today.